### ISSUE TITLE
Warnings not detected

### ISSUE THREAD

**Issue by siddux** [JacobDomagala/CompileResult#28]
Hey, I'm trying to use this action in my pipeline but it's not working as expected. It creates a comment saying that it has not detected any error nor warning but, checking the logs I get this:

```
Original string:
 --- stderr: gui
/home/ocu/ws/src/gui/src/helpers/rtsp_stream.cpp: In member function ‘void RTSPStream::handle_media_player_error(QMediaPlayer::Error)’:
/home/ocu/ws/src/gui/src/helpers/rtsp_stream.cpp:[15](https://github.com/siddux/pipeline_test/actions/runs/8204270022/job/22439203338#step:8:16)7:64: warning: unused parameter ‘error’ [-Wunused-parameter]
  157 | void RTSPStream::handle_media_player_error(QMediaPlayer::Error error)
      |                                            ~~~~~~~~~~~~~~~~~~~~^~~~~
In file included from /home/ocu/ws/src/gui/src/gui.cpp:5:
/home/ocu/ws/src/gui/include/gui/gui.hpp: In constructor ‘GUI::GUI(QApplication*, rclcpp::Node::SharedPtr, QWidget*)’:
/home/ocu/ws/src/gui/include/gui/gui.hpp:[17](https://github.com/siddux/pipeline_test/actions/runs/8204270022/job/22439203338#step:8:18)5:27: warning: ‘GUI::ros_node’ will be initialized after [-Wreorder]
  175 |   rclcpp::Node::SharedPtr ros_node;
```

So, why all of these warnings are not published in the comments? Thanks.

**Comment by JacobDomagala** [comment]
Can I see the .yaml file you use for the action and the output of the `CompileResult`? Maybe it's failing on something else? Is this a private repo?

**Comment by siddux** [comment]
Here you have the part of the yaml. Basically it gets a docker image built in a previous step, copies the file from it and then calls your action.

```yaml
warnings-bot:
    needs: build
    runs-on: ubuntu-22.04
    services:
      docker:
        image: docker:stable
        options: --privileged
    steps:
      - 
        name: Checkout code
        uses: actions/checkout@v3
      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
      -
        name: Download artifact
        uses: actions/download-artifact@v2
        with:
          name: ocu
          path: /tmp
      - 
        name: Load Docker image
        run: |
          docker load --input /tmp/ocu.tar
        working-directory: /tmp
      - 
        name: Copy file from Docker container to host
        run: |
          docker cp $(docker create ocu:latest):/home/ocu/ws/output.txt /tmp/output.txt
        working-directory: /tmp
      -
        name: Post PR comment for warnings/errors
        if: always()
        uses: JacobDomagala/CompileResult@master
        with:
          comment_title: UBUNTU COMPILE RESULT
          compile_result_file: /tmp/output.txt
          token: ${{ secrets.ACCESS_TOKEN }}
          compiler: CLANG
``` 
Here you have the results of the action:
[8_Post PR comment for warnings_errors.txt](https://github.com/JacobDomagala/CompileResult/files/14539849/8_Post.PR.comment.for.warnings_errors.txt)

Yes, this is a private repo. I've had some problems configurin the access token as it's not clear which permissions does it needs but I think I've sorted it.

**Comment by JacobDomagala** [comment]
You should try with setting `work_dir` action input (since you're not building within the same docker)
```yaml
        name: Post PR comment for warnings/errors
        if: always()
        uses: JacobDomagala/CompileResult@master
        with:
          comment_title: UBUNTU COMPILE RESULT
          compile_result_file: /tmp/output.txt
          token: ${{ secrets.ACCESS_TOKEN }}
          compiler: CLANG
          work_dir: /home/ocu/ws
```

**Comment by siddux** [comment]
That's it, adding  the `work_dir` is the solution when you work on a different environment than build one. Thanks!
